pkg/volume/gce_pd/gce_util.go

```diff
@@ -122,6 +112,135 @@ func (util *GCEDiskUtil) DetachDisk(pd *gcePersistentDisk) error {
 	if err := gce.(*gce_cloud.GCECloud).DetachDisk(pd.pdName); err != nil {
 		return err
 	}
+
+	return waitForDiskToDetach(pd)
+}
+
+// Wait for the disk device to be created
+func waitForDiskToAttach(pd *gcePersistentDisk, sdBeforeSet util.StringSet) (string, error) {
+	devicePaths := getDiskByIdPaths(pd)
+	for numTries := 0; numTries < 10; numTries++ {
+		if err := udevadmChangeToNewDrives(sdBeforeSet); err != nil {
+			// udevadm errors should not block disk attachment, log and continue
+			glog.Errorf("%v", err)
+		}
+
+		for _, path := range devicePaths {
```

Comment:
getDeviceByIdPaths() returns multiple symlinks to the same device file, but we only actually need/use one of them (/dev/disk/by-id/*).  Does it make sense to check for the existence of all of the other, unnecessary links here, or is that just inviting unnecessary failure?  I'm not sure.